pkg/objstore/azure/azure.go

```diff
@@ -0,0 +1,293 @@
+package azure
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+
+	blob "github.com/Azure/azure-storage-blob-go/2018-03-28/azblob"
+	"github.com/improbable-eng/thanos/pkg/objstore"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	opObjectsList  = "ListBucket"
+	opObjectInsert = "PutObject"
+	opObjectGet    = "GetObject"
+	opObjectHead   = "HeadObject"
+	opObjectDelete = "DeleteObject"
+)
+
+// Config Azure storage configuration.
+type Config struct {
+	StorageAccountName string `yaml:"storage-account"`
+	StorageAccountKey  string `yaml:"storage-account-key"`
+	ContainerName      string `yaml:"container"`
+}
+
+// Bucket implements the store.Bucket interface against s3-compatible APIs.
+type Bucket struct {
+	logger       log.Logger
+	containerURL blob.ContainerURL
+	config       *Config
+	opsTotal     *prometheus.CounterVec
+	closer       blobReadCloser
```

Comment:
I think this is broken.. Are you using this?

Reply:
What is broken here? closer is used in Close() function which is part of interface, as far as I know.